### PR TITLE
change rand() and mt_rand() to random_int()

### PR DESCRIPTION
**Pull request by teiling88**
### 1. Why is this change necessary?

image

### 6. Checklist

- [ ] I have written tests and verified that they fail without my change
- [ ] I have squashed any insignificant commits
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [ ] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: engine/Library/Zend/Mail.php [hunk 1]
```diff
@@ -1087,7 +1087,7 @@ public function createMessageId() {
             $user = getmypid();
         }
 
-        $rand = mt_rand();
+        $rand = random_int();
```
Comment (tinect) [inline engine/Library/Zend/Mail.php:1090]: there are 2 parameters needed 😱

File: engine/Library/Zend/Mail/Transport/File.php [hunk 2]
```diff
@@ -126,6 +126,6 @@ protected function _sendMail()
      */
     public function defaultCallback($transport)
     {
-        return 'ZendMail_' . $_SERVER['REQUEST_TIME'] . '_' . mt_rand() . '.tmp';
+        return 'ZendMail_' . $_SERVER['REQUEST_TIME'] . '_' . random_int() . '.tmp';
```
Comment (tinect) [inline engine/Library/Zend/Mail/Transport/File.php:129]: there are 2 parameters needed 😱

File: engine/Library/Zend/Mail.php (hunk 1 above)
Comment (teiling88) [reply engine/Library/Zend/Mail.php:1090]: upsi

File: engine/Library/Zend/Mail/Transport/File.php (hunk 2 above)
Comment (teiling88) [reply engine/Library/Zend/Mail/Transport/File.php:129]: upsi

Comment (jenskueper) [reply engine/Library/Zend/Mail/Transport/File.php:129]: Shopware has a random helper library implemented. You could use https://github.com/shopware/shopware/blob/5.5/engine/Shopware/Components/Random.php
instead. If something changes in the future, we only have to change it once.

**Pull request closed without merging**